src/core/zowe/core_for_zowe_sdk/zosmf_profile2.py

```diff
@@ -0,0 +1,302 @@
+        for (key, value) in secure_props.items():
+            if re.match(
+                "profiles\\." + base_profile_name + "\\.properties\\.[a-z]+", key
+            ):
+                property_name = key.split(".")[3]
+                base_props[property_name] = value
+
+        return base_props
+
+    def load(
+        self,
+        profile_name: Union[str, None] = None,
+        profile_type: Union[str, None] = None,
+        profile_args: Union[dict, None] = None,
+    ) -> dict:
+        """Load z/OSMF connection details from a z/OSMF profile.
+
+        Returns
+        -------
+        zosmf_connection
+            z/OSMF connection object
+
+        We will be loading properties from a bottom up fashion,
+        the bottom being the base/default profile properties
+        and the up being the explicitly mentioned Profile.
+
+        Loading Order :
+            Base Profile Properties
+        Overriding Order:
+            Service Profile (profile explicitly mentioned) properties
+            Profile args
+        """
+
+        # load config file
+        if self._config_dir is None:
+            self.autodiscover_config_dir()
+
+        self._config_filepath = os.path.join(self._config_dir, self._config_filename)
+
+        with open(self._config_filepath, encoding="UTF-8", mode="r") as fileobj:
+            profile_jsonc = jsonc.load(fileobj)
+
+        # load profile
+        if profile_type is None and profile_name is None:
+            raise ValueError(
+                "Both profile_type and profile_name cannot be empty at the same time."
+            )
+
+        service_profile = {}
+
+        # load base profile
+        base_profile: dict = self.load_base_profile_properties(
+            profile_jsonc=profile_jsonc
+        )
+        service_profile.update(base_profile)
+
+        # load given profile
+        if profile_name is None:
+            profile_name = self.get_profilename_from_profiletype(
+                profile_jsonc=profile_jsonc, profile_type=profile_type
+            )
+
+        required_profile: dict = self.load_profile_properties(
+            profile_jsonc=profile_jsonc, profile_name=profile_name
+        )
+        service_profile.update(required_profile)
+
+        # apply profile args
+        if profile_args:
+            service_profile.update(profile_args)
+
+        return service_profile
+
+    def load_credentials(self) -> dict:
+        """
+        return credentials stored for the given config
+        """
+        credentials: dict = {}
+
+        try:
+            secure_config = keyring.get_password(
+                constants["ZoweServiceName"], constants["ZoweUsername"]
+            )
+        except Exception as exc:
+            raise SecureProfileLoadFailed(
+                constants["ZoweServiceName"], error_msg=str(exc)
+            ) from exc
+
+        secure_config_json = jsonc.loads(base64.b64decode(secure_config).decode())
```

Comment:
This code fails on Windows where there is some platform-specific logic required for loading credentials:
https://github.com/zowe/zowe-client-python-sdk/blob/c2725e47555f99ba46edbc5d380ad38a6df99ca0/src/core/zowe/core_for_zowe_sdk/zosmf_profile.py#L100-L106

* The service name must have the account name added to the end like this: `{serviceName}/{accountName}`.
* The secret value must be decoded with UTF-16 before it can be Base64 decoded.

Reply:
hey, I've fixed the issue, could you test this on Windows?

Reply:
Thanks, I can confirm this works now to load credentials on Windows.

I got a unit tests failure in `tests/unit/test_zowe_core.py` related to this. I believe adding the following snippet in that file after `SECURE_CONFIG_PROPS` is defined should fix it:
```
        if sys.platform == "win32":
            SECURE_CONFIG_PROPS = SECURE_CONFIG_PROPS.decode("utf-16")
```

We may also need to do further testing with the keyring on Linux but that could be handled in a separate issue. I forget what OS you're using, if you're on Linux and it's already working for you then maybe we're already covered 🙂

Reply:
Yes, its working on Linux ☺️